frontend/src/hooks/board/useMenuActions.ts

```
@@ -0,0 +1,182 @@
+    setCanvasMode(CanvasMode.Selection);
+  },
+  [MenuActions.AddRectangle]: ({ canvas, properties, setCanvasMode }) => {
+    if (!canvas || !properties || !setCanvasMode) {
+      return;
+    }
+    addRectangle(canvas, properties);
+    setCanvasMode(CanvasMode.Selection);
+  },
+  [MenuActions.AddCircle]: ({ canvas, properties, setCanvasMode }) => {
+    if (!canvas || !properties || !setCanvasMode) {
+      return;
+    }
+    addCircle(canvas, properties);
+    setCanvasMode(CanvasMode.Selection);
+  },
+  [MenuActions.AddText]: ({ canvas, properties, setCanvasMode }) => {
+    if (!canvas || !properties || !setCanvasMode) {
+      return;
+    }
+    handleAddText(canvas, 50, 50, properties);
+    setCanvasMode(CanvasMode.Selection);
+  },
+  [MenuActions.GroupSelected]: ({ canvas }) => {
+    if (!canvas) {
+      return;
+    }
+    handleGroupSelected(canvas);
+  },
+  [MenuActions.RemoveSelected]: ({ canvas }) => {
+    if (!canvas) {
+      return;
+    }
+    handleRemoveSelected(canvas);
+  },
+  [MenuActions.ClearCanvas]: ({ canvas }) => {
+    if (!canvas) {
+      return;
+    }
+    canvas.clear();
+  },
+  [MenuActions.LoadCanvas]: ({ canvas, setCanvasMode }) => {
+    if (!canvas || !setCanvasMode) {
+      return;
+    }
+    handleLoadFromJSON(canvas);
+    setCanvasMode(CanvasMode.Selection);
+  },
+  [MenuActions.AddImageUrl]: ({ setCanvasMode }) => {
+    if (!setCanvasMode) {
+      return;
+    }
+    setCanvasMode(CanvasMode.Selection);
+  },
+  [MenuActions.AddImageDisk]: ({ setCanvasMode }) => {
+    if (!setCanvasMode) {
+      return;
+    }
+    setCanvasMode(CanvasMode.Selection);
+  },
+  [MenuActions.LoadImagesJson]: ({ setCanvasMode }) => {
+    if (!setCanvasMode) {
+      return;
+    }
+    setCanvasMode(CanvasMode.Selection);
+  },
+  [MenuActions.ChangeColor]: () => {
+    // No-op: handled by color picker UI
+  },
+  [MenuActions.ChangeSize]: () => {
+    // No-op: handled by size picker UI
+  },
```

I'm thinking if we should throw an error here so that we know that we should not use these functions.

---

okay, good point, i will do it